INTITA/Modules/LogInScreen/LogInViewController.swift

```diff
@@ -90,83 +101,117 @@ class LogInViewController: UIViewController, Storyboarded, AlertAcceptable {
+            cell = logoCell
+            
+        case .emailTextFieldCell:
             guard let emailCell = tableView.dequeueReusableCell(withIdentifier: "reuseForText") as? TextTableViewCell else { return UITableViewCell() }
             let cellConfig = TextTableViewCellConfiguration(type: .email, placeholderText: "inputEmail".localized)
             emailCell.configure(with: cellConfig)
-            return emailCell
-        case 2:
+            cell = emailCell
+            
+        case .passwordTextFieldCell:
             guard let passwordCell =  tableView.dequeueReusableCell(withIdentifier: "reuseForText") as? TextTableViewCell else { return UITableViewCell() }
             
             let cellConfig = TextTableViewCellConfiguration(type: .password, placeholderText: "inputPassword".localized)
             passwordCell.configure(with: cellConfig)
-            return passwordCell
-        case 3:
+            cell = passwordCell
+            
+        case .linksButtonCell:
             guard let linksCell = tableView.dequeueReusableCell(withIdentifier: "reuseForLinks") as? LinksTableViewCell else { return UITableViewCell() }
             linksCell.delegate = self
-            return linksCell
-        case 4:
+            cell = linksCell
+            
+        case .loginButtonCell:
             guard let buttonCell = tableView.dequeueReusableCell(withIdentifier: "reuseForButton") as? RegisterButtonTableViewCell else { return UITableViewCell() }
             buttonCell.delegate = self
-            return buttonCell
+            buttonCell.logInButton.setTitle("logIn".localized, for: .normal)
+            cell = buttonCell
+            
         default:
             return UITableViewCell()
         }
+        return cell ?? UITableViewCell()
     }
 }
 
 extension LogInViewController: RegisterButtonTableViewCellDelegate {
+    
     func didPressLogInButton(_ sender: RegisterButtonTableViewCell) {
-        guard let emailCell = tableView.cellForRow(at: IndexPath(row: 1, section: 0)) as? TextTableViewCell else { return }
-        guard let passwordCell = tableView.cellForRow(at: IndexPath(row: 2, section: 0)) as? TextTableViewCell else { return }
+        
+        guard let emailCell = tableView.cellForRow(at: IndexPath(row: 2, section: 0)) as? TextTableViewCell else { return }
+        guard let passwordCell = tableView.cellForRow(at: IndexPath(row: 4, section: 0)) as? TextTableViewCell else { return }
         
         emailCell.textField.resignFirstResponder()
         passwordCell.textField.resignFirstResponder()
         
-        guard let password = passwordCell.textField.text, let email = emailCell.textField.text else {
-            return
-        }
+        guard let password = passwordCell.textField.text, let email = emailCell.textField.text else { return }
         
-        if !validator.validateEmail(email: email) {
+        if (!validator.validateEmail(email: email)) && (email != "") {
```

Comment:
`if (!validator.validateEmail(email: email)) && (email != "") { ...`
Please replace logic into `validateEmail` function. 
Please do something like this: `if validator.isValidEmail(email) { ...`

Reply:
validator could return the array of errors [.emptyEmail, .wrongPassword].
 You can improve it in another pull

Reply:
It has done.